src/inference/graph_inference.cpp

```diff
@@ -341,6 +342,24 @@ class GraphNodeAssignment : public Nice2Assignment {
     return label_set_->GetLabelName(label_id);
   }
 
+  // Returns conditional probability of a label conditionated on its neighbouring labels
+  double GetNodeConditionalProbabilityGivenNeighbours(const GraphInference& fweights, int node, const double normalization, int label) const {
+    return exp(GetNodeScoreGivenAssignmentToANode(fweights, node, node, label)) / normalization;
+  }
+
+  // Returns the normalization related to a certain node
+  double GetNodeNormalizationConstantGivenNeighbours(const GraphInference& fweights, int node, std::vector<int>* candidates) const {
```

Comment:
I see some vector being appended. This method does not only compute normalization.